src/GafferOSL/ShadingEngine.cpp

```diff
@@ -970,13 +970,9 @@ static const T *varyingValue( const IECore::CompoundData *points, const char *na
 ShadingEngine::ShadingEngine( const IECoreScene::ShaderNetwork *shaderNetwork )
 	:	m_hash( shaderNetwork->Object::hash() ), m_timeNeeded( false ), m_unknownAttributesNeeded( false ), m_hasDeformation( false )
 {
-	ShaderNetworkPtr networkCopy;
-	if( true ) /// \todo Make conditional on OSL < 1.10
-	{
-		networkCopy = shaderNetwork->copy();
-		IECoreScene::ShaderNetworkAlgo::convertOSLComponentConnections( networkCopy.get() );
-		shaderNetwork = networkCopy.get();
-	}
+	ShaderNetworkPtr networkCopy = shaderNetwork->copy();
```

Comment:
> I did try the todo to avoid the conversion entirely, but that failed

Yep, that makes sense. OSL has a different convention for specifying component connections - we use `a.r` and they use `a[0]`. So even though `convertOSLComponentConnections()` isn't needed to convert shaders any more, it is still needed to convert between those conventions. The "Make conditional" comment was just inaccurate.